ruby/ql/lib/codeql/ruby/controlflow/internal/ControlFlowGraphImplShared.qll

```diff
@@ -853,6 +855,23 @@ private module Cached {
    */
   cached
   ControlFlowElement getAControlFlowExitNode(ControlFlowElement cfe) { last(cfe, result, _) }
+
+  /**
+   * Gets the CFG scope of node `n`. Unlike `getCfgScope`, this predicate
+   * is calculated based on reachability from an entry node, and it may
+   * yield different results for AST elements that are split into multiple
+   * scopes.
+   */
```

Comment:
Would it be possible to give a minimal example of code that would cause `getCfgScope` and `getNodeCfgScope` to give different results? I ran a brief Ruby query and couldn't find any cases, but maybe there's a simple one in C#?

Reply:
There are no examples for Ruby, but for C# we inline field initializations into the constructors, so that means that each initializer assignment may occur in multiple scopes. Example:

```csharp
class C
{
    public int Foo = 0;
    public C()
    {
    }

    public C(int i)
    {
        this.Foo += i;
    }
}
```

gets turned into

```csharp
class C
{
    public int Foo;
    public C()
    {
        Foo = 0;
    }

    public C(int i)
    {
        Foo = 0;
        this.Foo += i;
    }
}
```